workflows/temporal/temporal.go

```diff
@@ -106,42 +113,52 @@ func (c *TemporalWF) Terminate(ctx context.Context, req *workflows.WorkflowRefer
 	return nil
 }
 
-func (c *TemporalWF) Get(ctx context.Context, req *workflows.WorkflowReference) (*workflows.StateResponse, error) {
+func (c *TemporalWF) Get(ctx context.Context, req *workflows.GetRequest) (*workflows.StateResponse, error) {
 	c.logger.Debugf("getting workflow data")
 	resp, err := c.client.DescribeWorkflowExecution(ctx, req.InstanceID, "")
 	if err != nil {
 		return nil, err
 	}
+
+	var createdAtTime time.Time
+	if resp.WorkflowExecutionInfo.StartTime != nil {
+		createdAtTime = *resp.WorkflowExecutionInfo.StartTime
+	}
+
 	// Build the output struct
 	outputStruct := workflows.StateResponse{
-		WFInfo:    workflows.WorkflowReference{InstanceID: req.InstanceID},
-		StartTime: resp.WorkflowExecutionInfo.StartTime.Format(time.RFC3339),
-		Metadata: map[string]string{
-			"task_queue": resp.WorkflowExecutionInfo.GetTaskQueue(),
-			"status":     lookupStatus(resp.WorkflowExecutionInfo.Status),
+		Workflow: &workflows.WorkflowState{
+			InstanceID:    req.InstanceID,
+			CreatedAt:     createdAtTime,
+			LastUpdatedAt: createdAtTime,
+			RuntimeStatus: lookupStatus(resp.WorkflowExecutionInfo.Status),
+			Properties: map[string]string{
+				"task_queue": resp.WorkflowExecutionInfo.GetTaskQueue(),
```

Comment:
Should this use camelCase too? Unless it needs to use snake_case for compatibility with some external service

Reply:
It can be any form that the component implementation wants. For this Temporal component, the actual form doesn't matter, as long as it's documented. I was just copying what we had previously for v1.10 (which didn't actually ship with this), but it can be changed. In fact, if we want to change it, I might even namespace it to something like `temporalio.taskQueue` instead of `task_queue` or `taskQueue` to make it more clear that it's a Temporal-specific property.